spec/features/admin/reports_spec.rb

```diff
@@ -307,11 +308,11 @@
       product2.set_property 'Organic', 'NASAA 12345'
       product1.taxons = [taxon]
       product2.taxons = [taxon]
-      variant1.update_column(:count_on_hand, 10)
+      variant1.count_on_hand = 10
       variant1.update_column(:sku, "sku1")
-      variant2.update_column(:count_on_hand, 20)
+      variant2.count_on_hand = 20
       variant2.update_column(:sku, "sku2")
-      variant3.update_column(:count_on_hand, 9)
```

Comment:
This was persisting the changes to DB. Isn't it needed? I hope so!

Reply:
variant_stock.count_on_hand goes around and ends up persisting the stock item count on hand as well.

I am not sure yet but with this comment you made me realize I must have left a hole in my variant_override solution for v2. There's no vo.count_on_hand= setter... so, if a Variant is scoped and count_on_hand= is used, the stock item of the variant is still updated.
Basically, ScopeVariantToHub is missing overrides for on_hand= and on_demand=
I'll investigate and create an issue if required.
This is not related to this PR.

Reply:
today I came back to this very big question in v2. I have the answer.
**Why we don't need scope_variant_to_hub.count_on_hand= nor scope_variant_to_hub.on_demand= ?**
scope_variant_to_hub is what is added to the variant objects so that when methods are called, it will do what needs to be done on the VOs.

These are the multiple places where these would be needed and why they are not needed:
1 - on **checkout** cart_controller, cart_service will go through order_contents.add that will add variant to the line_item, the stock movements are done in order_inventory that uses stock_location and, through overriding in OFN, we make it go through variant.move that is overridden in scope_variant_to_hub and updates vo.count_on_hand
2 - in **order edit page** we use api/shipment that also uses order_contents.add, same process as above
3 - in the **product edit page** we go through admin/products/bulk_update that uses product_set that sets variant.on_hand and variant.on_demand directly, here we are not supposed to use overrides
4 - in **inventory edit page**, we use admin/variant_overrides/bulk_update that uses VariantOverrideSet that will set on_hand and on_demand directly on the VariantOverride columns, so, no scope_variant_to_hub is needed

This is why no on_hand=, count_on_hand= or on_demand= are NOT present in scope_variant_to_hub
Only scope_variant_to_hub.move is needed which is used in point 1 and 2 above.

**Summary, it's all good, nothing needs to be done.** ;-)